### PR TITLE
cgroup2: port over eBPF device controller from crun

### LINKED ISSUE #2144: cgroup2: eBPF device controller
https://github.com/AkihiroSuda/runc/commits/ebpf

I'm trying to port over [`libcrun/ebpf.c` from crun v0.10.2](https://github.com/containers/crun/blob/0.10.2/src/libcrun/ebpf.c), which is licensed under LGPL3 (or later) and incompatible with Apache License 2.0

@giuseppe Do you think my code should be treated as a derivative of your C code? If so, any chance to relicense `ebpf.c` and `cgroup.c` as Apache License 2.0?

### PR DESCRIPTION
**Pull request by AkihiroSuda**
The implementation is based on https://github.com/containers/crun/blob/0.10.2/src/libcrun/ebpf.c

Although `ebpf.c` is originally licensed under LGPL-3.0-or-later, the author Giuseppe Scrivano agreed to relicense the file in Apache License 2.0: https://github.com/opencontainers/runc/issues/2144#issuecomment-543116397

See `libcontainer/cgroups/ebpf/devicefilter/devicefilter_test.go` for tested configurations.

Close #2144

### CODE REVIEW

**Comment by AkihiroSuda** [comment]
```
CGO_ENABLED=1 GOARCH=arm GOARM=6 CC=arm-linux-gnueabi-gcc go build -buildmode=pie  -ldflags "-X main.gitCommit="bf9519326d3dcc4a78f3cddbc54ac7a78a0aa948" -X main.version=1.0.0-rc9+dev " -tags "seccomp apparmor selinux ambient" -o runc-armel .
# github.com/opencontainers/runc/vendor/github.com/cilium/ebpf
vendor/github.com/cilium/ebpf/syscalls.go:285:17: constant 3405662737 overflows int32
Makefile:125: recipe for target 'localcross' failed
```

**Comment by AkihiroSuda** [comment]
@giuseppe LGTY?

File: libcontainer/cgroups/fs/devices_v2.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+	}
+	if len(cgroup.Resources.DeniedDevices) != 0 {
+		// never set by OCI specconv
+		return errors.New("libcontainer DeniedDevices is not supported, use Devices")
+	}
+	insts, license, err := devicefilter.DeviceFilter(cgroup.Devices)
+	if err != nil {
+		return err
+	}
+	logrus.Debugf("eBPF device filter (%s)\n%s", license, insts.String())
```
Comment (crosbymichael) [inline libcontainer/cgroups/fs/devices_v2.go]: can we remove the debug line now?

Comment (AkihiroSuda) [reply libcontainer/cgroups/fs/devices_v2.go]: removed

**Comment by crosbymichael** [comment]
LGTM

**Comment by AkihiroSuda** [comment]
rebased

**Comment by AkihiroSuda** [comment]
@dqminh @hqhq PTAL? This is the blocker for supporting cgroup2.

**Comment by mrunalp** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**